### ISSUE TITLE
frontmatter handling

### ISSUE THREAD

**Issue by noraj** [markdownlint/markdownlint#442]
## Describe the Enhancement:

Add an option to take frontmatter into consideration.

### Impacted Rules:

- MD002
- MD032
- MD041

## Describe the Need:

For most SSG (static site generator), you have the ability to add frontmatters (front matter; front-matter) at the beginning of your markdown files.

For example, to add [tags to a page in retype](https://retype.com/configuration/page/#tags), it's most often YAML frontmatters. But some other SSG like Hexo also support JSON [frontmatters](https://hexo.io/docs/front-matter).

So if one adds YAML frontmatters to a markdown file like that:

```yaml
---
title: Hello World
date: 2013/7/13 20:46:25
---
# Markdown title

Some **Markdown** content.
```

it will trigger the following rules:

```
test.md:11: MD002 First header should be a top level header
test.md:3: MD032 Lists should be surrounded by blank lines
test.md:5: MD032 Lists should be surrounded by blank lines
test.md:1: MD041 First line in file should be a top level header
```

just because markdownlint don't take into consideration the frontmatter and so generate false positives.

Even if it's not enabled by default, having an option to tell markdownlint files are using YAML or JSON frontmatters would be nice.

## Current Alternative

The dirty workaround is to disable the impacted rules.

**Comment by noraj** [comment]
PS : I searched for [frontmatter](https://github.com/markdownlint/markdownlint/issues?q=is%3Aissue+is%3Aopen+frontmatter+) and there was 0 issue, but in fact searching for [front matter](https://github.com/markdownlint/markdownlint/issues?q=is%3Aissue+is%3Aopen+front+matter+) there were existing issues.

**Comment by noraj** [comment]
Solution is to use `--ignore-front-matter`  option in the CLI (found in #247) or to add `ignore-front-matter true` in `.mdlrc` (found in #260).

cf. https://github.com/markdownlint/markdownlint/blob/main/docs/configuration.md#general-options
